# Final report notebook 

**This notebook will show insights, tables and charts about the starwars api(*swapi*) data**

## Tables 
- Tables used in this notebook are generated on the analysis notebook using the data stored on the database(swapi.db).
- Check the **analysis.ipynb** notebook for more info about the tables used.

## Charts 
- Charts used in this notebook are generated on the visualization notebook using tables from the analysis notebook and new tables generated also from the database(swapi.db).
- Check the **visualization.ipynb** for more info about the charts used. 

In [6]:
import pandas as pd

In [4]:
# Creating a function to speed up the process of connecting to the csv tables 
def csv (table) : 
    return pd.read_csv(f"../Output/tables/{table}.csv") 

### Tallest people 

In [7]:
tallest_ppl = csv("tallest_ppl")
tallest_ppl.head(5) 

,name,height
0,Yarael Poof,264
1,Tarfful,234
2,Lama Su,229
3,Chewbacca,228
4,Roos Tarpals,224


<img src="../Output/charts/tallest_ppl.png" style="max-width:100%; height:auto;">

> As we can see in the output, The tallest person is **Yarael Poof** with a height of *264* 

### Gender status (people)

<img src="../Output/charts/gender_status.png" style="max-width:100%; height:auto;">

> As we can see in the chart, The most popular gender between humans is **"male"**

### Average height per gender (people table)

<img src="../Output/charts/gender_avg_height.png" style="max-width:100%; height:auto;">

>  As we can see in the output, The tallest gender by avg height is **"none"**

### Heaviest people by mass

In [9]:
heaviest_ppl = csv("heaviest_ppl")
heaviest_ppl.head(5) 

,name,mass
0,Jabba Desilijic Tiure,1358.0
1,Grievous,159.0
2,IG-88,140.0
3,Darth Vader,136.0
4,Tarfful,136.0


<img src="../Output/charts/heaviest_ppl.png" style="max-width:100%; height:auto;">

> As we can see in the output, The person with the highest mass is **Jabba Desilijic Tiure** with a mass of *1358.0* 

### Oldest people 

In [10]:
oldest_ppl = csv('oldest_ppl')
oldest_ppl.head(5) 

,name,birth_year,birth_year_num
0,Yoda,896BBY,896
1,Jabba Desilijic Tiure,600BBY,600
2,Chewbacca,200BBY,200
3,C-3PO,112BBY,112
4,Dooku,102BBY,102


<img src="../Output/charts/oldest_ppl.png" style="max-width:100%; height:auto;">

> As we can see in the output, The oldest person is **Yoda** with an age of *896 Before the Battle of Yavin (869BBY)* 

### Planets with the highest population 

In [11]:
planets_population = csv("planets_population")
planets_population.head(5) 

,name,population
0,Coruscant,1000000000000
1,Skako,500000000000
2,Geonosis,100000000000
3,Mon Cala,27000000000
4,Eriadu,22000000000


<img src="../Output/charts/planets_population.png" style="max-width:100%; height:auto;">

> As we can see in the output, The planet with the highest population is **Coruscant** with a population of *1000000000000* 

### Planets with the highest surface_water

In [15]:
planets_water = csv("planets_water") 
planets_water.head(5) 

,name,surface_water
0,Kamino,100.0
1,Hoth,100.0
2,Mon Cala,100.0
3,Ojom,100.0
4,Bestine IV,98.0


<img src="../Output/charts/planets_water.png" style="max-width:100%; height:auto;">

> As we can see in the output table, The planets with the highest water_surface are : **Kamino, Hoth, Mon Cala and Ojom** with a water surface of *100.0*

### Planets with the most films 

In [16]:
top_planets_by_films = csv("top_planets_by_films") 
top_planets_by_films.head(5) 

,name,film_count
0,Tatooine,5.0
1,Naboo,4.0
2,Coruscant,4.0
3,Dagobah,3.0
4,Alderaan,2.0


> As we can see in the output table, The planet with the most films is **Tatooine** with *5* films

### Most frequent **species** in films 

In [17]:
frequent_species = csv("frequent_species") 
frequent_species.head(5) 

,name,film_count
0,Human,6
1,Droid,6
2,Yoda's species,5
3,Twi'lek,4
4,Wookie,4


> As we can see in the output table, The most frequent species is **Humans** and **Droid** with 6 films 

### People with the most starships

In [19]:
top_people_by_starships = csv("top_people_by_starships") 
top_people_by_starships.head(3) 

,name,starships_count
0,Obi-Wan Kenobi,5
1,Anakin Skywalker,3
2,Padmé Amidala,3


> As we can see in the output table, The person with the most starships is **Obi-Wan Kenobi** with *5 starships*

### People with the most vehicles 

In [21]:
top_people_by_vehicles = csv("top_people_by_vehicles")
top_people_by_vehicles.head(3) 

,name,vehicles_count
0,Luke Skywalker,2
1,Anakin Skywalker,2
2,Leia Organa,1


> As we can see in the output table, The person with the most vehicles is **Luke Skywalker** and **Anakin Skywalker** with *2 vehicles*

### Tallest species 

In [23]:
tallest_sp = csv("tallest_sp") 
tallest_sp.head(5) 

,name,average_height
0,Hutt,300
1,Quermian,240
2,Kaminoan,220
3,Wookie,210
4,Toong,200


> As we can see in the output chart, The tallest species is the **Hutt** with an average height of 300 